core/framework/extractors/tree_sitter_extractor_backup_20250913.py

```
@@ -0,0 +1,945 @@
+        # Detect language from file extension
+        language = self._detect_language(file_path)
+        if not language:
+            logger.debug(f"No Tree-sitter parser for {file_path.suffix}")
+            return self._empty_result()
+
+        # Load content if not provided
+        if content is None:
+            try:
+                with open(file_path, encoding='utf-8', errors='ignore') as f:
+                    content = f.read()
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)